server/sandbox_run.go

```diff
@@ -491,6 +502,8 @@ func (s *Server) RunPodSandbox(ctx context.Context, req *pb.RunPodSandboxRequest
 		return nil, err
 	}
 
+	s.addContainer(container)
```

Comment:
Is this intentional?

Reply:
Yeah, otherwise it won't show up in the inspect endpoint.

Reply:
:confused: sounds like a bug we had for long :(

Reply:
Not really, as we didn't want to expose the infra container out. ListContainers doesn't care for it AFAICT. This is a special case for making cadvisor happy.